wpiunits/src/main/java/edu/wpi/first/units/Measure.java

```diff
@@ -98,6 +98,24 @@ default U baseUnit() {
     return (U) unit().getBaseUnit();
   }
 
+  /**
+   * Absolute value of measure.
+   *
+   * @return a new measure containing the result
+   */
+  default Measure<U> abs() {
+    return baseUnitMagnitude() > 0 ? this : times(-1);
+  }
+
+  /**
+   * Take the sign of another measure.
+   * @param other measure from which to take sign
+   * @return a new measure with the sign from another measure
+   */
+  default Measure<U> copySign(Measure<U> other) {
```

Comment:
This should be called signum() to match the Java standard library.

Reply:
This is slightly different than `signum()`. `Math.signum()` simply returns +1, 0 or -1 based on the sign of the input. This method copies the sign of the input variable, but maintains the magnitude.
Its similar to `Math.copySign()`.
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#copySign-double-double-

Reply:
This function returns the wrong result if `this` is nonzero and `other` is zero. It inverts the sign when it shouldn't since 1 != 0.

Reply:
Hmm, I get an error when I try to use `Math.copySign()`.
`Type mismatch: cannot convert from Measure<capture#1-of ?> to Measure<U>`

```
/**
   * Take the sign of another measure.
   * @param other measure from which to take sign
   * @return a new measure with the sign from another measure
   */
  default Measure<U> copySign(Measure<U> other) {
    return baseUnit().of(Math.copySign(this.baseUnitMagnitude(), other.baseUnitMagnitude()));
  }
  ```
  
  But if I do
  `return this.times(other).baseUnitMagnitude() >= 0.0 ? this : times(-1);`
  its fine. Which should be fine, if the sign of both this and other are the same the product will be positive, and if they're different the sign needs to be flipped.

Reply:
`Unit.of` returns a wildcarded `Measure<?>` because the self-type isn't known. You'd need to cast to `Measure<U>`.

Another issue is nonlinear units. What's the absolute value of -10°C? Using the base unit (Kelvin), it's still -10°C (263.15K); but using the relative unit Celsius it'd be 10°C, which would be treated as 283.15K in any math operations

Reply:
Perhaps `abs()` should also take in the desired Unit?
Also, how would I resolve an unchecked cast for the wildcard `Measure<?>` ?